tchannel/tchannel.py

```diff
@@ -135,6 +146,9 @@ def call(self, scheme, service, arg1, arg2=None, arg3=None,
             transport.SCHEME: scheme,
             transport.CALLER_NAME: self.name,
         }
+        if shard_key:
```

Comment:
oh man... this debt is getting out of hand...

we really need to move to allowing a `tchannel.Request` to call soon...